test/e2e/aws_test.go

```diff
@@ -118,6 +140,172 @@ var _ = Describe("functional tests", func() {
 	})
 })
 
+func scaleMachineDeployment(namespace, machineDeployment string, replicasCurrent int32, replicasProposed int32) {
+	fmt.Fprintf(GinkgoWriter, "Scaling MachineDeployment from %d to %d\n", replicasCurrent, replicasProposed)
+	deployment := &clusterv1.MachineDeployment{}
+	Expect(kindClient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: namespace, Name: machineDeployment}, deployment))
+	deployment.Spec.Replicas = &replicasProposed
+	Expect(kindClient.Update(context.TODO(), deployment)).NotTo(HaveOccurred())
+	waitForMachinesCountMatch(namespace, machineDeployment, replicasCurrent, replicasProposed)
+	waitForMachineDeploymentRunning(namespace, machineDeployment, replicasProposed)
+}
+
+func createMachineTemplate(namespace, clusterName, machineDeploymentName, awsMachineTemplateName, bootstrapConfigName string,nodepool string, replicas int32) {
+	fmt.Fprintf(GinkgoWriter, "Creating MachineDeployment in namespace %s with name %s\n", namespace, machineDeploymentName)
+	makeAWSMachineTemplate(namespace, awsMachineTemplateName)
+	makeJoinBootstrapConfigTemplate(namespace, bootstrapConfigName)
+	makeMachineDeployment(namespace, machineDeploymentName, awsMachineTemplateName, bootstrapConfigName, clusterName, nodepool, replicas)
+	waitForMachinesCountMatch(namespace, machineDeploymentName,replicas, replicas)
+	waitForMachineDeploymentRunning(namespace, machineDeploymentName, replicas)
+}
+
+func waitForMachinesCountMatch(namespace, machineDeploymentName string, replicasCurrent int32, replicasProposed int32) {
```

Comment:
Should be able to use the value from the retrieved MachineDeployment here rather than passing in `replicasCurrent`

Reply:
From testing perspective, i want to ensure that the MachineDeployment is getting updated properly. Hence, not depending on MachineDeployment.